WooCommerce/Classes/ViewRelated/Orders/Order Details/OrderDetailsViewController.swift

```diff
@@ -439,14 +439,13 @@ private extension OrderDetailsViewController {
             }
             collectPaymentTapped()
         case .reprintShippingLabel(let shippingLabel):
-            guard let navigationController = navigationController else {
-                assertionFailure("Cannot reprint a shipping label because `navigationController` is nil")
-                return
-            }
+            let printNavigationController = WooNavigationController()
             let coordinator = PrintShippingLabelCoordinator(shippingLabels: [shippingLabel],
                                                             printType: .reprint,
-                                                            sourceNavigationController: navigationController)
-            coordinator.showPrintUI()
+                                                            sourceNavigationController: printNavigationController)
+            let printViewController = coordinator.createPrintViewController()
+            printNavigationController.viewControllers = [printViewController]
+            present(printNavigationController, animated: true)
```

Comment:
I moved it out because it's presented from both here, and the Shipping Label purchase flow, and the way it needs to be presented is different from the two places. Because that's an implementation detail of the context it's presented from, it makes it more sense to me to present it in those contexts, rather than the coordinator.